### PR TITLE
[14147] Fix documentation installation

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

## Description

CMake option `BUILD_DOCUMENTATION` downloads using `wget` the Readthedocs ZIP file from an URL that does not have the latest documentation. Changing to this other URL solves the problem: https://fast-dds.docs.eprosima.com/_/downloads/en/v2.6.0/htmlzip/, though instead of using `wget`, `curl` must be used.

I have tested it in Ubuntu 20.04. I suggest the reviewer to check that it also works as expected in Windows. To test this PR run:

```
colcon build --packages-select fastrtps --cmake-args -DBUILD_DOCUMENTATION=ON
```

In case of using a `colcon.meta` file, ensure that CMake option `CHECK_DOCUMENTATION` is disabled (by default CMake option `EPROSIMA_BUILD` sets the later to `ON`).

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
* *N/A* The code follows the style guidelines of this project. 
* *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
* *N/A* Any new/modified methods have been properly documented using Doxygen. 
* *N/A* Fast DDS test suite has been run locally. 
* *N/A* Changes are ABI compatible. 
* *N/A* Changes are API compatible. 
* *N/A* Documentation builds and tests pass locally. 
* *N/A* New feature has been added to the `versions.md` file (if applicable).
* *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by EduPonz — APPROVED** [review]
LGTM, I've successfully tested in Ubuntu 20.04. I'll wait for @inesluno's Windows review/test before merging

**Comment by inesluno** [comment]
I have successfully tested in Windows.

**Pull request merged**
